shared/src/main/scala/mlscript/TyperDatatypes.scala

```diff
@@ -234,7 +234,7 @@ abstract class TyperDatatypes extends TyperHelpers { self: Typer =>
         else TopType
       subst(td.kind match {
         case Als => td.bodyTy
-        case Nms => throw new Exception("Namespaces are not supported yet.")
+        case Nms => assert(false, "Namespaces are not supported yet."); td.bodyTy
```

Comment:
Same here. `assert(false, ...)` is quite smelly.

Reply:
Actually, the more logical thing to throw here would rather be `new NotImplementedError(...)`.

Reply:
Sorry for the back and forth 😂

Assertions are for things that should never happen if the code is correct. Not the same as something not yet implemented.

Reply:
Updated.